power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -767,7 +957,22 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
         }
     }
 
-    auto pilot_run(std::vector<std::vector<RegulatedTransformer>> const& regulator_order) const {
+    void update_binary_search(std::vector<std::vector<RegulatedTransformer>> const& regulator_order) const {
+        for (std::size_t i = 0; i < regulator_order.size(); ++i) {
+            const auto& sub_order = regulator_order[i];
+            for (std::size_t j = 0; j < sub_order.size(); ++j) {
```

Comment:
here and in other places: we do have `using size_t` in `common.hpp`

```suggestion
            for (size_t j = 0; j < sub_order.size(); ++j) {
```

Reply:
but not two of them

Reply:
comment addressed in the latest commit.